pxf/pxf-ignite/src/main/java/org/apache/hawq/pxf/plugins/ignite/IgniteResolver.java

```diff
@@ -0,0 +1,173 @@
+package org.apache.hawq.pxf.plugins.ignite;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.hawq.pxf.api.OneRow;
+import org.apache.hawq.pxf.api.OneField;
+import org.apache.hawq.pxf.api.io.DataType;
+import org.apache.hawq.pxf.api.utilities.ColumnDescriptor;
+import org.apache.hawq.pxf.api.utilities.InputData;
+import org.apache.hawq.pxf.api.ReadResolver;
+import org.apache.hawq.pxf.api.WriteResolver;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.text.SimpleDateFormat;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.codec.binary.Hex;
+
+import com.google.gson.JsonArray;
+
+
+
+/**
+ * PXF-Ignite resolver class
+ */
+public class IgniteResolver extends IgnitePlugin implements ReadResolver, WriteResolver {
+    private static final Log LOG = LogFactory.getLog(IgniteResolver.class);
+    
+    // HAWQ column descriptors
+    private ArrayList<ColumnDescriptor> columns = null;
+
+    public IgniteResolver(InputData input) throws Exception {
+        super(input);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Constructor started");
+        }
+
+        columns = input.getTupleDescription();
+        
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Constructor successful");
+        }
+    }
+
+    /**
+     * Transform a JsonArray object stored in {@link OneRow} into a List {@link OneField}
+     */
+    @Override
+    public List<OneField> getFields(OneRow row) throws Exception {
+        JsonArray result = (JsonArray)row.getData();

```

Comment:
just wondering - does this line generate a warning? if so, we should make a safer cast or suppress the warning.